hal/inc/hal_cellular_global_identity.h

```diff
@@ -0,0 +1,12 @@
+#ifndef HAL_CELLULAR_GLOBAL_IDENTITY_H
+#define HAL_CELLULAR_GLOBAL_IDENTITY_H
+
+typedef struct
+{
+    uint16_t mobile_country_code;
```

Comment:
This fields is this struct are defined by the 3GPP. It should be akin to a phone number `<country_code>-<area_code>-<prefix>-<customer>`.

I don't expect it to change too much, but I will check the web to see if a version is applied to these fields and sizes.

Reply:
This was a suggestion for ABI stability as we do for most of the structs nowadays in case we ever want to extend them and not in the sense that there are multiple variants defined by 3GPP.

Reply:
Actually, I would insist on this change.

Reply:
How big should the version member be and what should the number scheme look like?

Reply:
In the current iteration we have a `schema` field for describing the data format, as well as a `size` field.
At this time, I think we should choose a single 16-bit field. It would allow for better packing of the data, which would provide better space efficiency as well as faster data access.
As it stands, we are currently wasting 24-bits, and force aligning the `schema` field.